src/corehost/cli/exe/exe.cmake

```diff
@@ -0,0 +1,64 @@
+# Copyright (c) .NET Foundation and contributors. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+cmake_minimum_required (VERSION 2.6)
+project (${DOTNET_HOST_EXE_NAME})
+
+if(WIN32)
+    add_compile_options($<$<CONFIG:RelWithDebInfo>:/MT>)
+    add_compile_options($<$<CONFIG:Release>:/MT>)
+    add_compile_options($<$<CONFIG:Debug>:/MTd>)
+else()
+    add_compile_options(-fPIE)
+endif()
+
+include(../../setup.cmake)
+
+# Include directories
+if(WIN32)
+    include_directories("${CLI_CMAKE_RESOURCE_DIR}/${DOTNET_HOST_EXE_NAME}")
+endif()
+include_directories(../../../)
+include_directories(../../../common)
+include_directories(../../)
+include_directories(../../fxr)
+
+# CMake does not recommend using globbing since it messes with the freshness checks
+set(SOURCES
+    ../../../corehost.cpp
+    ../../../common/trace.cpp
+    ../../../common/utils.cpp)
+
+
+if(${DOTNET_HOST_EXE_NAME} MATCHES "apphost")
```

Comment:
You can move these to respective cmake files for apphost and dotnet, alleviating the need for this conditional to exist at all.